### PR TITLE
Fixes for Precise images

### PR DESCRIPTION
**Pull request by emdantrim**

### CODE REVIEW

File: ci_environment/bison/recipes/default.rb [hunk 1]
```diff
@@ -25,24 +25,27 @@
...
+     action   :install
+     source   path
+     provider Chef::Provider::Package::Dpkg
 
-    notifies :delete, resources(:file => path)
+     notifies :delete, resources(:file => path)
+   end
+  package "bison" do
+    action [:install, :upgrade]
   end
```
Comment (meatballhat) [inline ci_environment/bison/recipes/default.rb]: I'm confused about what's intended here 😃.  Is the default precise version of `bison` sufficient, and, if so, can all of the dpkg download + install stuff be removed?

File: ci_environment/redis/recipes/default.rb [hunk 2]
```diff
@@ -21,12 +21,12 @@
...
-apt_repository "rwky-redis" do
-  uri          "http://ppa.launchpad.net/rwky/redis/ubuntu"
+apt_repository "chris-lea-redis-server" do
+  uri          "http://ppa.launchpad.net/chris-lea/redis-server/ubuntu"
   distribution node['lsb']['codename']
   components   ["main"]
-  key          "5862E31D"
-  keyserver    "hkp://ha.pool.sks-keyservers.net"
+  key          "136221EE520DDFAF0A905689B9316A7BC7917B12"
+  keyserver    "hkp://keyserver.ubuntu.com"
```
Comment (meatballhat) [inline ci_environment/redis/recipes/default.rb]: If the key we need is available via the HA pool, I'd like to stick with that keyserver ❤️

File: ci_environment/bison/recipes/default.rb (hunk 1 above)
Comment (emdantrim) [inline ci_environment/bison/recipes/default.rb]: this is what I had done previously, but @Lyoness reverted it later on. @Lyoness, what do you think?

Comment (Lyoness) [inline ci_environment/bison/recipes/default.rb]: Lines 46-49 can be deleted.  I didn't delete all in the revert. 

the bison binary filename was fixed in attributes. 
https://github.com/travis-ci/travis-cookbooks/pull/842/commits/538831f10ecc36f642bdae383fb44e870635e026

File: ci_environment/redis/recipes/default.rb (hunk 2 above)
Comment (Lyoness) [inline ci_environment/redis/recipes/default.rb]: I cannot find the chris-lea-redis-server key anywhere on the ha.pool

**Comment by emdantrim** [comment]
Tested in image build and didn't blow up. This is ready to go whenever.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
Does this change mean we no longer support hhvm?

**Pull request merged**